tutorial/20_recipes/006_user_defined_pruner.py

```diff
@@ -0,0 +1,127 @@
+gradient descent classifier (``SGDClassifier``) run on the sklearn iris dataset. We
+implement a pruner which terminates a trial at a certain step if it is in last
+place compared to completed trials at the same step. We begin considering
+pruning after a "warmup" of 1 training step and 5 completed trials. For
+demonstration purposes, we print() a diagnostic message from prune() when it is
+about to return True (indicating pruning).
+
+It may be important to note that the SGDClassifier score, as it is evaluated on
+a holdout set, decreases with enough training steps due to overfitting. This
+means that a trial could be pruned even if it had a favorable (high) value on a
+previous training set. After pruning, Optuna will take the intermediate value
+last reported as the value of the trial.
+
+"""
+
+import numpy as np
+from sklearn.datasets import load_iris
+from sklearn.model_selection import train_test_split
+from sklearn.linear_model import SGDClassifier
+
+import optuna
+from optuna.pruners import BasePruner
+from optuna.trial._state import TrialState
+
+
+class LastPlacePruner(BasePruner):
+    def __init__(self, warmup_steps, warmup_trials):
+        self._warmup_steps = warmup_steps
+        self._warmup_trials = warmup_trials
+
+    def prune(self, study: "optuna.study.Study", trial: "optuna.trial.FrozenTrial") -> bool:
+        # Get the latest score reported from this trial
+        step = trial.last_step
+
+        if step:  # trial.last_step == None when no scores have been reported yet
+            this_score = trial.intermediate_values[step]
+
+            # Get scores from other trials in the study reported at the same step
+            completed_trials = study.get_trials(deepcopy=False, states=(TrialState.COMPLETE,))
+            other_scores = [
+                t.intermediate_values[step]
+                for t in completed_trials
+                if step in t.intermediate_values
+            ]
+            other_scores = sorted(other_scores)
+
+            # Prune if this trial at this step has a lower value than all completed trials
+            # at the same step. Note that steps will begin numbering at 0 in the objective
+            # function definition below.
+            if step >= self._warmup_steps and len(other_scores) > self._warmup_trials:
+                if this_score < other_scores[0]:
+                    print(f"prune() True: Trial {trial.number}, Step {step}, Score {this_score}")
+                    return True
+
+        return False
+
+
+###################################################################################################
+# Lastly, let's confirm the implementation is correct with the simple hyperparameter optimization.
+
+def objective(trial):
+    iris = load_iris()
+    classes = np.unique(iris.target)
+    X_train, X_valid, y_train, y_valid = train_test_split(
+        iris.data, iris.target, train_size=100, test_size=50, random_state=0)
```

Comment:
Could you fix the CI failure?
```suggestion
        iris.data, iris.target, train_size=100, test_size=50, random_state=0
    )
```